File: app/src/main/java/org/michaelbel/moviemade/rest/model/v3/Collection.java
package org.michaelbel.moviemade.rest.model.v3;

import com.google.gson.annotations.SerializedName;

import org.michaelbel.tmdb.v3.json.Movie;
import org.michaelbel.tmdb.TmdbObject;

import java.util.List;

public class Collection extends TmdbObject {

    @SerializedName("id")
    public int id;

    @SerializedName("name")
    public String name;

    @SerializedName("overview")
    public String overview;

    @SerializedName("poster_path")
    public String posterPath;

    @SerializedName("backdrop_path")
    public String backdropPath;

    @SerializedName("parts")
    public List<Movie> movies;

    public Collection() {}
}

File: moviemade/src/main/java/org/michaelbel/moviemade/app/data/model/Collection.kt
package org.michaelbel.moviemade.app.data.model

import com.google.gson.annotations.SerializedName

data class Collection(
    @SerializedName("id") val id: Int = 0,
    @SerializedName("name") val name: String? = null,
    @SerializedName("overview") val overview: String? = null,
    @SerializedName("poster_path") val posterPath: String? = null,
    @SerializedName("backdrop_path") val backdropPath: String? = null,
    @SerializedName("parts") val parts: List<Movie> = emptyList()
)